config/routes.rb

```diff
@@ -0,0 +1,9 @@
+Rails.application.routes.draw do
+  devise_for :users
+  root :to => "posts#index"
+  resources :categories
+  resources :posts do
+    resources :tags
```

Comment:
Thinking we do not need for tags to be a nested resource of posts. Or feel that we can limit the routes that are exposed. Something like `resources :tags, only: [:create, :destroy]`